public/app/plugins/datasource/loki/formatterTS.ts

```diff
@@ -0,0 +1,406 @@
+    const value = query.substring(valueNode.from, valueNode.to);
+
+    if (ipLabelFilterNode.getChild(Eq)) {
+      return `| ${identifier}=ip(${value})`;
+    } else if (ipLabelFilterNode.getChild(Neq)) {
+      return `| ${identifier}!=ip(${value})`;
+    }
+  }
+
+  if (node.getChild(UnitFilter) && node.getChild(UnitFilter)!.getChild(DurationFilter)) {
+    const subNode = node.getChild(UnitFilter)!;
+    const durationFilterNode = subNode.getChild(DurationFilter)!;
+
+    const identifierNode = durationFilterNode.getChild(Identifier)!;
+    const valueNode = durationFilterNode.getChild(Duration)!;
+    const identifier = query.substring(identifierNode.from, identifierNode.to);
+    const value = query.substring(valueNode.from, valueNode.to);
+
+    if (durationFilterNode.getChild(Gtr)) {
+      return `| ${identifier}>${value}`;
+    } else if (durationFilterNode.getChild(Gte)) {
+      return `| ${identifier}>=${value}`;
+    } else if (durationFilterNode.getChild(Lss)) {
+      return `| ${identifier}<${value}`;
+    } else if (durationFilterNode.getChild(Lte)) {
+      return `| ${identifier}<=${value}`;
+    } else if (durationFilterNode.getChild(Neq)) {
+      return `| ${identifier}!=${value}`;
+    } else if (durationFilterNode.getChild(Eq)) {
+      return `| ${identifier}=${value}`;
+    } else if (durationFilterNode.getChild(Eql)) {
+      return `| ${identifier}==${value}`;
+    }
+  }
+
+  if (node.getChild(UnitFilter) && node.getChild(UnitFilter)!.getChild(BytesFilter)) {
+    const subNode = node.getChild(UnitFilter)!;
+    const bytesFilterNode = subNode.getChild(BytesFilter)!;
+
+    const identifierNode = bytesFilterNode.getChild(Identifier)!;
+    const valueNode = bytesFilterNode.getChild(Bytes)!;
+    const identifier = query.substring(identifierNode.from, identifierNode.to);
+    const value = query.substring(valueNode.from, valueNode.to);
+
+    if (bytesFilterNode.getChild(Gtr)) {
+      return `| ${identifier}>${value}`;
+    } else if (bytesFilterNode.getChild(Gte)) {
+      return `| ${identifier}>=${value}`;
+    } else if (bytesFilterNode.getChild(Lss)) {
+      return `| ${identifier}<${value}`;
+    } else if (bytesFilterNode.getChild(Lte)) {
+      return `| ${identifier}<=${value}`;
+    } else if (bytesFilterNode.getChild(Neq)) {
+      return `| ${identifier}!=${value}`;
+    } else if (bytesFilterNode.getChild(Eq)) {
+      return `| ${identifier}=${value}`;
+    } else if (bytesFilterNode.getChild(Eql)) {
+      return `| ${identifier}==${value}`;
+    }
+  }
+
+  if (node.getChild(NumberFilter)) {
+    const numberFilterNode = node.getChild(NumberFilter)!;
+
+    const identifierNode = numberFilterNode.getChild(Identifier)!;
+    const valueNode = numberFilterNode.getChild(Number)!;
+    const identifier = query.substring(identifierNode.from, identifierNode.to);
+    const value = query.substring(valueNode.from, valueNode.to);
+
+    if (numberFilterNode.getChild(Gtr)) {
```

Comment:
Here you could also use something like `nextSibling` and avoid the literal check.